tests/utils/test_msa_aligner.py

```diff
@@ -0,0 +1,145 @@
+from unittest import TestCase
+from unittest.mock import patch, Mock, mock_open
+import os
+from make_prg.utils.msa_aligner import MAFFT, NotAValidExecutableError, ExecutionError
+from pathlib import Path
+import subprocess
+import filecmp
+from Bio import SeqIO
+from tests.test_helpers import make_alignment
+
+
+class TestMAFFT(TestCase):
+    def setUp(self) -> None:
+        self.executable = "fake_exec"
+        self.tmpdir = Path("fake_path")
+
+    @patch("shutil.which", return_value=None)
+    def test___constructor___invalid_executable(self, shutil_which_mock):
+        with self.assertRaises(NotAValidExecutableError):
+            MAFFT(self.executable)
+
+        shutil_which_mock.assert_called_once_with(self.executable, mode=os.X_OK)
+
+    @patch("shutil.which", return_value=1)
+    @patch.object(Path, Path.mkdir.__name__)
+    def test___constructor___valid(self, mkdir_mock, shutil_which_mock):
+        mafft = MAFFT(self.executable, self.tmpdir)
+
+        self.assertEqual(self.executable, mafft._executable)
+        self.assertEqual(self.tmpdir, mafft._tmpdir)
+        mkdir_mock.assert_called_once_with(parents=True, exist_ok=True)
+        shutil_which_mock.assert_called_once_with(self.executable, mode=os.X_OK)
+
+    @patch("shutil.which", return_value=1)
+    @patch.object(Path, Path.mkdir.__name__)
+    @patch("subprocess.Popen")
+    def test___run_aligner___ok_run(self, popen_mock, *uninteresting_mocks):
+        wait_mock = Mock()
+        wait_mock.wait.return_value = 0
+        popen_mock.return_value = wait_mock
+        args = "fake_arg_1 fake_arg_2 fake_arg_3"
+        env = {"fake_key_1": "fake_value_1", "fake_key_2": "fake_value_2"}
+
+        mafft = MAFFT(self.executable, self.tmpdir)
+        mafft._run_aligner(args, env)
+
+        popen_mock.assert_called_once_with(args, stderr=subprocess.PIPE, encoding="utf-8", shell=True, env=env)
+        wait_mock.wait.assert_called_once_with()
+
+    @patch("shutil.which", return_value=1)
+    @patch.object(Path, Path.mkdir.__name__)
+    @patch("subprocess.Popen")
+    def test___run_aligner___fail_run(self, popen_mock, *uninteresting_mocks):
+        wait_mock = Mock()
+        wait_mock.wait.return_value = 1
```

Comment:
may be worth commenting, or stating in function name, that code 1 signals failure, and that's whats being tested

Reply:
Done in https://github.com/iqbal-lab-org/make_prg/pull/34/commits/2ef3adb012a9214ae1f5207e1cf07b4a6a48c09f